src/popover.jsx

```diff
@@ -17,30 +17,34 @@ export default class Popover extends React.Component {
     this.state = {
       topOffset: 0,
       leftOffset: 0,
-      isOpen: false
+      isOpen: props.isOpen || false
     };
   }
 
-  componentDidMount() {
-    this.calculateHeights();
+  componentWillReceiveProps(props){
+    if(props.isOpen != this.state.isOpen) this.setState({isOpen: props.isOpen})
+  }
 
-    document.addEventListener('click', (ev) => {
-      ev.stopPropagation();
-      this.handleClick(true, ev);
-    });
+  componentDidMount() {
+    if(this.state.isOpen) this.calculateHeights();
```

Comment:
I am doing the willReceiveProps for if you want to pass isOpen as a prop
for that one use case. Also, we can't calculate the height because it's not
rendering if it's hidden since that's an antipattern really.
On Thu, Sep 3, 2015 at 16:51 Darin Haener <notifications@github.com> wrote:

> In src/popover.jsx
> <https://github.com/dphaener/reactable-popover/pull/1#discussion_r38709948>
> :
>
> >
> > -    document.addEventListener('click', (ev) => {
> > -      ev.stopPropagation();
> > -      this.handleClick(true, ev);
> > -    });
> > +  componentDidMount() {
> > +    if(this.state.isOpen) this.calculateHeights();
>
> Also, I think it should calculate the dimensions regardless of whether it
> is open or not.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dphaener/reactable-popover/pull/1/files#r38709948>.
>